src/router.ts

```diff
@@ -1,25 +1,40 @@
-const router = require("express").Router();
-const { expressjwt } = require('express-jwt');
-const userController = require("./controllers/userController")
+import express from 'express';
+import { expressjwt } from 'express-jwt';
+import { generalConfig } from './config/generalConfig';
+import { signinController } from './controllers/signinController';
+import { signupController } from './controllers/signupController';
+import { oauthServer, castRequest, castResponse } from './middleware/oauthServer';
 
+const router = express.Router();
+
+// TODO: Make this configurable and pass in as ENV variable
 const secret = process.env.JWTSECRET;
 
 const authMiddleware = expressjwt({
-    algorithms: ['HS256'],
-    credentialsRequired: false,
-    secret,
+  algorithms: ['HS256'],
+  credentialsRequired: false,
+  secret: generalConfig.jwtSecret as string,
 });
 
+// Support for email+password
+router.post('/signin', (req, res) => signinController(req, res));
+router.post('/signup', (req, res) => signupController(req, res));
+
+// Support for OAuth2
+router.get('/authorize', (req, res) => oauthServer.authorize(castRequest(req), castResponse(res)));
```

Comment:
One thing I'm not clear about is how the frontend will get the Client ID. Does it request it from the backend server?

Reply:
It won't need to use this route. It will use the sign in/up routes

Reply:
I did some reading about the Client ID and Client Secret yesterday, and it seems that the frontend would get it from the OAuth provider and store it somewhere safe.